deeppavlov/core/models/tf_model.py

```diff
@@ -13,119 +13,97 @@
+                               ' have sess attribute!'.format(self.__class__.__name__))
+        super().__init__(*args, **kwargs)
 
-    def infer(self, instance, *args, **kwargs):
-        """
-        Just a wrapper for a private method.
-        """
-        return self._forward(instance, *args, **kwargs)
+    def load(self):
+        """Load model parameters from self.load_path"""
+        path = str(self.load_path.resolve())
+        # Check presence of the model files
+        if tf.train.checkpoint_exists(path):
+            log.info('[loading model from {}]'.format(path))
+            # Exclude optimizer variables from saved variables
+            var_list = [var for var in tf.trainable_variables()
+                        if not var.name.startswith('Optimizer')]
+            saver = tf.train.Saver(var_list)
+            saver.restore(self.sess, path)
 
     def save(self):
-        save_path = str(self.save_path)
-        saver = tf.train.Saver()
-        log.info('[saving model to {}]'.format(save_path))
-        saver.save(self.sess, save_path)
-        log.info('model saved')
-
-    def get_checkpoint_state(self):
-        if self.load_path:
-            try:
-                if self.load_path.parent.is_dir():
-                    return tf.train.get_checkpoint_state(self.load_path.parent)
-                else:
-                    raise ConfigError
-            except ConfigError:
-                log.error('Provided `load_path` is incorrect!', exc_info=True)
-                sys.exit(1)
-        else:
-            log.warning('No `load_path` is provided for {}'.format(self.__class__.__name__))
-
-    @overrides
-    def load(self):
+        """Save model parameters to self.save_path"""
+        path = str(self.save_path.resolve())
+        log.info('[saving model to {}]'.format(path))
+        var_list = [var for var in tf.trainable_variables()
+                    if not var.name.startswith('Optimizer')]
+        saver = tf.train.Saver(var_list)
+        saver.save(self.sess, path)
+
+    def get_train_op(self, loss, learning_rate, optimizer=None, clip_norm=None, learnable_scopes=None):
+        """ Get train operation for given loss
+
+        Args:
+            loss: loss, tf tensor or scalar
+            learning_rate: scalar or placeholder
+            clip_norm: clip gradients norm by clip_norm
+            learnable_scopes: which scopes are trainable (None for all)
+            optimizer: instance of tf.train.Optimizer, default Adam
+
+        Returns:
+            train_op
         """
-        Load session from checkpoint
+        with tf.variable_scope('Optimizer'):
+            if learnable_scopes is None:
+                variables_to_train = tf.trainable_variables()
+            else:
+                variables_to_train = []
+                for scope_name in learnable_scopes:
+                    for var in tf.trainable_variables():
+                        if var.name.startswith(scope_name):
```

Comment:
`var.name.startswith(scope_name)` -> `scope_name not in var.name`?

Reply:
done